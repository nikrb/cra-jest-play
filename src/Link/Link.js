import React from 'react';
import {StyleSheet, css} from 'aphrodite';

const STATUS = {
  NORMAL: 'normal',
  HOVERED: 'hovered'
};

export default class Link extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      class: STATUS.NORMAL
    }
  }
  mouseEnter = () => {
    this.setState( {class: STATUS.HOVERED});
  };
  mouseLeave = () => {
    this.setState( {class: STATUS.NORMAL})
  };
  isNormal = () => {
    return this.state.class === STATUS.NORMAL;
  };
  render = () => {
    // const classname = css(this.isNormal() ? styles.normal : styles.hovered);
    const classname = css( this.state.class===STATUS.NORMAL?styles.normal:styles.hovered);
    return (
      <a className={classname}
        href={this.props.page || "#"}
        onMouseEnter={this.mouseEnter}
        onMouseLeave={this.mouseLeave}
      >
        {this.props.children}
      </a>
    );
  };
};

const styles = StyleSheet.create({
  normal: {
    color: "blue"
  },
  hovered: {
    color: "red"
  }
});
