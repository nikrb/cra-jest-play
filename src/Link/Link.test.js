import React from 'react';
import Link from './Link';
import renderer from 'react-test-renderer';
import {StyleSheetTestUtils} from 'aphrodite';

beforeEach(() => {
  StyleSheetTestUtils.suppressStyleInjection();
});
afterEach(() => {
  StyleSheetTestUtils.clearBufferAndResumeStyleInjection();
});

test('Link change the class when hovered', done => {
  const component = renderer.create(
    <Link page='https://hissues.herokuapp.com'>HIssues</Link>
  )
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  tree.props.onMouseEnter();
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  const style = component.get(0).style;
  expect(style).toHaveProperty('color', 'red');

  tree.props.onMouseLeave();
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  return done();
});
