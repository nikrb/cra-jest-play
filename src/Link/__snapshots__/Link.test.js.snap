// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link change the class when hovered 1`] = `
<a
  className="normal_1tsdo2i"
  href="https://hissues.herokuapp.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  HIssues
</a>
`;

exports[`Link change the class when hovered 2`] = `
<a
  className="hovered_137u7ef"
  href="https://hissues.herokuapp.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  HIssues
</a>
`;

exports[`Link change the class when hovered 3`] = `
<a
  className="normal_1tsdo2i"
  href="https://hissues.herokuapp.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  HIssues
</a>
`;
